docs/namespaces.md

```diff
@@ -12,13 +12,410 @@ certainly want the docs that go with that version.</h1>
+
+Each user community wants to be able to work in isolation from other communities.
+
+Each user community has its own:
+
+1. resources (pods, services, replication controllers, etc.)
+2. policies (who can or cannot perform actions in their community)
+3. constraints (this community is allowed this much quota, etc.)
+
+A cluster operator may create a Namespace for each unique user community.
+
+The Namespace provides a unique scope for: 
+
+1. named resources (to avoid basic naming collisions)
+2. delegated management authority to trusted users
+3. ability to limit community resource consumption
+
+## Use cases
+
+1.  As a cluster operator, I want to support multiple user communities on a single cluster.
+2.  As a cluster operator, I want to delegate authority to partitions of the cluster to trusted users
+    in those communities.
+3.  As a cluster operator, I want to limit the amount of resources each community can consume in order
+    to limit the impact to other communities using the cluster.
+4.  As a cluster user, I want to interact with resources that are pertinent to my user community in
+    isolation of what other user communities are doing on the cluster.
+
+
+## Usage
+
+### Viewing namespaces
+You can list the current namespaces in a cluster using:
+```sh
+$> kubectl get namespaces
+NAME          LABELS    STATUS
+default       <none>    Active
+kube-system   <none>    Active
+```
+
+Kubernetes starts with two initial namespaces:
+   * ```default``` The default namespace for objects with no other namespace
+   * ```kube-system``` The namespace for objects created by the Kubernetes system
+
+You can also get the details about a specific namespace using:
+
+```
+kubectl get namespaces <name>
+```
+
+### Creating a new namespace
+
+To create a new namespace, first create a new YAML file called ```my-namespace.yaml``` with the contents:
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: <insert-namespace-name-here>
+spec:
+  finalizers:
+  - kubernetes
+```
+
+More information on the ```finalizers``` field can be found [below](#finalizers).
+
+Then run:
+```
+kubectl create -f my-namespace.yaml
+```
+
+### Setting the namespace for a request
+
+To temporarily set the namespace for a request, use the ```--namespace``` flag.
+
+For example:
+```
+kubectl --namespace=<insert-namespace-name-here> run nginx --image=nginx
+kubectl --namespace=<insert-namespace-name-here> get pods
+```
+
+### Setting the namespace preference
+
+You can permanently save the namespace for all subsequent kubectl commands in that
+context.
+
+First get your current context:
+```sh
+export CONTEXT=$(kubectl config view | grep current-context | awk '{print $2}')
+```
+
+Then update the default namespace:
+```sh
+kubectl config set-context $(CONTEXT) --namespace=<insert-namespace-name-here>
+```
+
+### Deleting a namespace
+
+You can delete a namespace with
```

Comment:
We may want to note that this is an asynchronous operation as content is being purged, the namespace will appear in the Terminating phase.